lib_eio_linux/eio_stubs.c

```diff
@@ -33,10 +34,16 @@ CAMLprim value caml_eio_mkdirat(value v_fd, value v_path, value v_perm) {
 CAMLprim value caml_eio_getrandom(value v_ba, value v_off, value v_len) {
   CAMLparam1(v_ba);
   ssize_t ret;
-  void *buf = Caml_ba_data_val(v_ba) + Long_val(v_off);
-  caml_enter_blocking_section();
-  ret = getrandom(buf, Long_val(v_len), 0);
-  caml_leave_blocking_section();
-  if (ret == -1) uerror("getrandom", Nothing);
+  size_t off = (size_t)Long_val(v_off);
+  size_t len = (size_t)Long_val(v_len);
+  while (off < len) {
+    void *buf = Caml_ba_data_val(v_ba) + off;
+    caml_enter_blocking_section();
+    ret = getrandom(buf, len - off, 0);
+    caml_leave_blocking_section();
+    if (ret < 0 && errno == EINTR) continue;
```

Comment:
Bikeshedding a bit, `I'd` suggest comparing explicitly with == -1, that's what caml_eio_mkdirat() above already does, just to keep the same style.

Reply:
off and len could be ssize_t as well, avoids a signed vs unsigned in off += ret.

Reply:
is `EINTR` == -1 always true? I am mostly following mirage-crypto style since that is what prompted this PR.
https://github.com/mirage/mirage-crypto/blob/main/rng/unix/mc_getrandom_stubs.c#L24

Reply:
I mean `ret`: ```if (ret == -1 && errno == EINTR) continue;```
manpage says:
`       On error, -1 is returned, and errno is set to indicate the error.`

Reply:
Ah, yes. You are correct. Incorporated in the last commit.